lib/resty/core/request.lua

```diff
@@ -310,16 +314,53 @@ do
 
         local rc
         if not value then
-            rc = C.ngx_http_lua_ffi_req_header_set_single_value(r, name,
-                                                         #name, nil, 0)
+            if not override then
+                error("invalid header value", 3)
+            end
+
+            rc = C.ngx_http_lua_ffi_req_set_header(r, name, #name, nil, 0, nil,
+                                                   0, 1)
 
         else
-            if type(value) ~= "string" then
-                value = tostring(value)
+            local sval, sval_len, mvals, mvals_len, buf
+
+            if type(value) == "table" then
+                mvals_len = #value
+                if mvals_len == 0 and not override then
+                    error("invalid header value", 3)
```

Comment:
Just like the `ngx_resp.add_header()` API, we should early-exit here, but not consider the argument invalid.